scipy/optimize/meson.build

```diff
@@ -159,8 +159,8 @@ __nnls = py3.extension_module('__nnls',
   install: true,
   subdir: 'scipy/optimize')
 
-res = run_command(py3, '-c', 'import os; print(os.environ.get("SCIPY_USE_PYTHRAN", 1))')
-if res.returncode() == 0
+res = run_command(py3, '-c', 'import os; print(os.environ.get("SCIPY_USE_PYTHRAN", 0))').stdout().strip()
```

Comment:
When set to `1`, this we get error like following,
```bash
dynamic module does not define module export function
```

Reply:
I'll have a look, doesn't ring a bell yet.

Reply:
Full traceback. Sometimes the `sparse` test work, sometimes they give cython signature error. This is a bit random. I am having a look into it.
```python
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/czgdp1807ssd/scipy_project/meson_build/install/lib/python3.9/site-packages/scipy/optimize/__init__.py", line 400, in <module>
    from .optimize import *
  File "/home/czgdp1807ssd/scipy_project/meson_build/install/lib/python3.9/site-packages/scipy/optimize/optimize.py", line 36, in <module>
    from ._numdiff import approx_derivative
  File "/home/czgdp1807ssd/scipy_project/meson_build/install/lib/python3.9/site-packages/scipy/optimize/_numdiff.py", line 8, in <module>
    from ._group_columns import group_dense, group_sparse
ImportError: dynamic module does not define module export function (PyInit__group_columns)
```

Reply:
I have sent a commit. It seems like some compiler flags were missing which I picked from the logs of `setup.py`. The above error hopefully is fixed.